azurerm/internal/services/mssqlvm/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,559 @@
+							ValidateFunc: validation.StringInSlice([]string{
+								string(sqlvirtualmachine.GENERAL),
+								string(sqlvirtualmachine.OLTP),
+								string(sqlvirtualmachine.DW),
+							}, false),
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMsSqlVirtualMachineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQLVM.SQLVirtualMachinesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+
+	id, _ := azure.ParseAzureResourceID(d.Get("virtual_machine_resource_id").(string))
+	name := id.Path["virtualMachines"]
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroupName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q): %+v", name, resourceGroupName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_sql_virtual_machine", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	sqlServerLicenseType := d.Get("sql_server_license_type").(string)
+	virtualMachineResourceName := d.Get("virtual_machine_resource_id").(string)
+
+	properties := sqlvirtualmachine.Properties{
+		SQLServerLicenseType:     sqlvirtualmachine.SQLServerLicenseType(sqlServerLicenseType),
+		VirtualMachineResourceID: &virtualMachineResourceName,
+	}
+
+	if sqlVirtualMachineGroupResourceID, ok := d.GetOk("sql_virtual_machine_group_resource_id"); ok {
+		SQLVirtualMachineGroupResourceID := sqlVirtualMachineGroupResourceID.(string)
+		properties.SQLVirtualMachineGroupResourceID = &SQLVirtualMachineGroupResourceID
+	}
+
+	if sqlImageSku, ok := d.GetOk("sql_image_sku"); ok {
+		SQLImageSku := sqlvirtualmachine.SQLImageSku(sqlImageSku.(string))
+		properties.SQLImageSku = SQLImageSku
+	}
+
+	if _, ok := d.GetOk("auto_patching_settings"); ok {
+		properties.AutoPatchingSettings = expandArmSqlVirtualMachineAutoPatchingSettings(d)
+	}
+
+	if _, ok := d.GetOk("key_vault_credential_settings"); ok {
+		properties.KeyVaultCredentialSettings = expandArmSqlVirtualMachineKeyVaultCredentialSettings(d)
+	}
+
+	if _, ok := d.GetOk("server_configurations_management_settings"); ok {
+		properties.ServerConfigurationsManagementSettings = expandArmSqlVirtualMachineServerConfigurationsManagementSettings(d)
+	}
+
+	if _, ok := d.GetOk("storage_configuration_settings"); ok {
+		properties.StorageConfigurationSettings = expandArmSqlVirtualMachineStorageConfigurationSettings(d)
+	}
+
+	Tags := d.Get("tags").(map[string]interface{})
+
+	parameters := sqlvirtualmachine.SQLVirtualMachine{
+		Location:   utils.String(location),
+		Properties: &properties,
+		Tags:       tags.Expand(Tags),
```

Comment:
we can do this inline 
```suggestion
		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
```